# Bioinformatic pipeline for ABBA-BABA analyses of genomic allele frequency data of _Sepsis cynipsea_ and _S. neocynipsea_ in Europe

#### (1) Trimming of raw reads, which are deposited on the SRA archive under [PRJNA612154](https://www.ncbi.nlm.nih.gov/bioproject?LinkName=biosample_bioproject&from_uid=14363816) was carried out with [trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic).

```bash
java -jar trimmomatic-0.36.jar \
PE -threads 20 \
input_R1.fastq.gz \
input_R2.fastq.gz \
output_R1.fastq.gz \
output_R1_un.fastq.gz \
output_R2.fastq.gz \
output_R2_un.fastq.gz \
ILLUMINACLIP:AdapterSeq_new.fa:2:30:7:5:true LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:40
```

#### (2) Mapping of the trimmed reads was carried out with bwa mem following the pipeline in [Kapun _et al._ 2020](https://academic.oup.com/mbe/article/37/9/2661/5837682) as described in [here](https://github.com/capoony/DrosEU_pipeline) using the reference genome of _S. thoracica_, which can be found [here](http://www.cgae.de/seto_01_genome_masked.fasta).

#### (3) After merging the BAM files with _samtools mpileup_ as described [here](https://github.com/capoony/DrosEU_pipeline), we called SNPs using the heuristic SNP caller [PoolSNP](https://github.com/capoony/PoolSNP) using the following parameters.

```bash
sh PoolSNP.sh  \
mpileup=/ABBA_BABA.mpileup.gz \
reference=seto_01_genome.fasta \
names=PhC,PtC,SoC,ZuC,MoC,GeN,HoN,SoN,MoN,IZuC,IZuN,Sor  \
max-cov=0.9 \
min-cov=10 \
min-count=20 \
min-freq=0.01 \
miss-frac=0 \
jobs=10 \
base-quality=15 \
output=ABBA_BABA \
badsites=0
```

#### (4) Then, we converted the VCF generated by PoolSNP to the sync file format using a script from [here](https://github.com/capoony/DrosEU_pipeline), removed all sites with missing information and calculated allele frequencies for the overall major allele using a script of our new ABBA-BABA test implementation, which can be found [here](https://github.com/nhmvienna/ABBABABA-4AF). The allele frequency file that was generated here can be found on Datadryad in [here](https://doi.org/10.5061/dryad.fxpnvx0sm)

```bash
gunzip -c ABBA_BABA.vcf.gz  | parallel \
  -k \
  --pipe \
  -j20 \
  --no-notice \
  --cat python /github/capoony/DrosEU_pipeline/VCF2sync.py \
  --input {} \
  | grep -v ".:.:.:.:.:." \
  | gzip > ABBA_BABA-filtered.sync.gz
```

```bash
gunzip -c ABBA_BABA-filtered.sync.gz \
  | parallel \
  -k \
  --pipe \
  -j 20 \
  --no-notice \
  --cat python /github/nhmvienna/ABBABABA-4AF/SYNC2AF.py \
  --sync {} \
  | gzip > Sepsid_ABBA-BABA.af.gz
```

#### (5) Finally, we conducted ABBA-BABA tests using the approaches of Durand _et al._ (2011) and Soraggi _et al._ (2018) using our new python implementation, which can be found here [here](https://github.com/nhmvienna/ABBABABA-4AF).

Note that the column positions that describe the populations used as P1,P2,P3 and P4 and are passed to the script via the parameter --Order are 0-based, i.e. 3 = 4th column. The first three columns in the input file are Chromosome, Position and allelic state, respectively, and all populations are arranged as PhC,PtC,SoC,ZuC,MoC,GeN,HoN,SoN,MoN,IZuC,IZuN,Sor.
Thus, the order 3,5,10,14  corresponds to: P1=PhC, P2=SoC, P3=SoN and P4=Sor. In the example below, _D_ will be calculated in windows comprised of 500 consecutive SNPs. Please check the [ABBABABA-4AF github page](https://github.com/nhmvienna/ABBABABA-4AF) for more details.

```bash
python3 /github/nhmvienna/ABBABABA-4AF/ABBABABA-4AF.py \
--AF Sepsid_ABBA-BABA.af.gz \
--SNPs 500 \
--Order 3,5,10,14 \
--Output output-file
```
